go-controller/pkg/testing/testifymockhelper.go

```diff
@@ -1,7 +1,60 @@
 package testing
 
+import (
+	"github.com/stretchr/testify/mock"
+)
+
+// TestifyMockHelper captures the arguments needed for the `On` , `Return` and 'Times` method ( refer
+// https://godoc.org/github.com/stretchr/testify/mock#Call.On ,
+// https://godoc.org/github.com/stretchr/testify/mock#Call.Return and
+// https://godoc.org/github.com/stretchr/testify/mock#Call.Times )
 type TestifyMockHelper struct {
-	OnCallMethodName    string
+	// OnCallMethodName - mock method that will be called.
+	// Refer the `Method` field at https://godoc.org/github.com/stretchr/testify/mock#Call
+	OnCallMethodName string
+	// OnCallMethodArgType - argument types of the method that will be called.
+	// Refer the `Arguments` field at https://godoc.org/github.com/stretchr/testify/mock#Call
 	OnCallMethodArgType []string
-	RetArgList          []interface{}
+	// RetArgList - arguments returned by mock method when called.
+	// Refer the `ReturnArguments` field at https://godoc.org/github.com/stretchr/testify/mock#Call
+	RetArgList []interface{}
+	// OnCallMethodsArgsStrTypeAppendCount - number of times the `string` type argument is repeated at the end.
+	// NOTE: There are a couple of cases where the ovn related calls take upto 23 string arguments
+	OnCallMethodsArgsStrTypeAppendCount int
+	// CallTimes - number of times to return the return arguments when mock method is called
+	// Refer the `Repeatability` field at https://godoc.org/github.com/stretchr/testify/mock#Call
+	// NOTE: The zero value for an int is 0 in go, however, it the value for 'CallTimes' is not set, we interpret that
+	// it will need to be called once.
+	CallTimes int
+}
+
+// PreProcessMockFnList allows for handling mocking of multiple/differnt method calls on a single mock object
+func PreProcessMockFnList(mockObj *mock.Mock, mArgs []TestifyMockHelper) {
```

Comment:
I think this name should be changed something like `ProcessMockFNList` or something more relevent to the user `ProcessExpectedMockedFNList()`

Another thought is it important that we mention the mocking aspect at all? `ProcessExpectedFNCallList()`

Reply:
Are you suggesting to drop the 'Pre' from the 'PreProcessMockFnList'? sure I could do that if it makes it more readable and conveys what the function does. 
We are using testify's Mock Struct and hence keeping the mock aspect.